### PR TITLE
Paddle

### PR DESCRIPTION
**Pull request by NMandel527**

### CODE REVIEW

File: src/main/java/bbm/brickbreaker/GameComponent.java [hunk 1]
```diff
@@ -32,6 +33,6 @@ protected void paintComponent(Graphics g) {
         g.fillOval((int) ball.getX() - 10, (int) ball.getY()  - 10, 20, 20);
 
         g.setColor(Color.BLUE);
-        g.fillRect(paddle.getX(),  paddle.getY(), paddle.getWidth(), paddle.getHeight());
+        g.fillRect((int) paddle.getX(), (int) paddle.getY(), (int) paddle.getWidth(), (int) paddle.getHeight());
```
Comment (AlizaBuchen) [inline src/main/java/bbm/brickbreaker/GameComponent.java:36]: double?

File: src/main/java/bbm/brickbreaker/Controller.java [hunk 2]
```diff
@@ -61,23 +60,45 @@ private void breakBricks() {
...
             }
         }
     }
 
-    private void checkPanelCollision() {
-        Rectangle panelBounds = new Rectangle(paddle.getX(), paddle.getY(), paddle.getWidth(), paddle.getHeight());
-        Rectangle ballBounds = new Rectangle((int) ball.getX(), (int) ball.getY(), radius * 2, radius * 2);
+    private void checkPaddleCollision() {
+        Rectangle paddleBounds = new Rectangle((int) paddle.getX(), (int) paddle.getY(),
+                (int) paddle.getWidth(), (int) paddle.getHeight());
```
Comment (AlizaBuchen) [inline src/main/java/bbm/brickbreaker/Controller.java:74]: can we use a double instead?

File: src/main/java/bbm/brickbreaker/GameFrame.java [hunk 3]
```diff
@@ -26,7 +27,8 @@ public GameFrame() {
         JLabel bar = new JLabel();
         bar.setOpaque(true);
         bar.setBackground(Color.BLUE);
-        bar.setBounds(paddle.getX(), paddle.getY(), paddle.getWidth(), paddle.getHeight());
+
+        bar.setBounds((int) paddle.getX(), (int) paddle.getY(), (int) paddle.getWidth(), (int) paddle.getHeight());
```
Comment (AlizaBuchen) [inline src/main/java/bbm/brickbreaker/GameFrame.java:31]: instead of casting everything to an int, can we use a double?

**Review by AlizaBuchen — APPROVED** [review]

File: src/main/java/bbm/brickbreaker/Controller.java [hunk 4]
```diff
@@ -80,12 +80,34 @@ private void breakBricks() {
...
-    private void checkPanelCollision() {
-        Rectangle panelBounds = new Rectangle(panel.getX(), panel.getY(), panel.getWidth(), panel.getHeight());
-        Rectangle ballBounds = new Rectangle((int) ball.getX(), (int) ball.getY(), radius * 2, radius * 2);
-
-        if (panelBounds.intersects(ballBounds)) {
-            ball.bounce(Bounds.TOP);
+    private void checkPaddleCollision() {
+        Rectangle paddleBounds = new Rectangle((int) paddle.getX(), (int) paddle.getY(),
+                (int) paddle.getWidth(), (int) paddle.getHeight());
+        Rectangle ballBounds = new Rectangle((int) ball.getX(), (int) ball.getY(),
```
Comment (cblitz2) [inline src/main/java/bbm/brickbreaker/Controller.java:75]: maybe make variables for ball.getX and ball.getY

**Review by cblitz2 — APPROVED** [review]

**Pull request merged**